pandas/core/arrays/datetimelike.py

```diff
@@ -437,6 +437,13 @@ def _with_freq(self, freq):
         arr._freq = freq
         return arr
 
+    def factorize(self, na_sentinel=-1):
```

Comment:
we don't want to expose factorize on Index at all.

Reply:
It is already there in the status quo.  If you want to remove it, that needs a deprecation cycle, is a separate issue

Reply:
```
/Users/jreback/pandas
bash-3.2$ grep -r factorize pandas/core/indexes/
pandas/core/indexes//multi.py:from pandas.core.arrays.categorical import factorize_from_iterables
pandas/core/indexes//multi.py:    indexer_from_factorized,
pandas/core/indexes//multi.py:        codes, levels = factorize_from_iterables(arrays)
pandas/core/indexes//multi.py:        codes, levels = factorize_from_iterables(iterables)
pandas/core/indexes//multi.py:        codes, uniques = algos.factorize(indexer, sort=True)
pandas/core/indexes//multi.py:            ok_codes, uniques = algos.factorize(indexer[mask], sort=True)
pandas/core/indexes//multi.py:            indexer = indexer_from_factorized(primary, primshp, compress=False)
Binary file pandas/core/indexes//__pycache__/multi.cpython-38.pyc matches
```

pls show where

Reply:
hmm i guess its in the base class.

In any event. I don't think we actually want to change this.

Reply:
Comment in existing tests seems to think this is already the behavior: https://github.com/pandas-dev/pandas/pull/33836/files#diff-8cf55ac38b6988b09a7f7f5d7280eb0fL360

Also should improve perf since this short-circuits the expensive part of factorize